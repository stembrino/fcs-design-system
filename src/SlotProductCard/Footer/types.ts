export interface FooterProps {
  btnColor: string;
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}
