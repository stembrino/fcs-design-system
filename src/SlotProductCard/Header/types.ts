export interface HeaderProps {
  heartColor: string;
  like: boolean;
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}
