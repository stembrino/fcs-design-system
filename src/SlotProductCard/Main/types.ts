export interface MainProps {
  imageSrc: string;
  backgroundSVGColor: string;
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}
