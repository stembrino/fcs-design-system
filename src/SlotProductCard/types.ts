export interface SlotProductCardProps {
  src: string;
  color?: string;
  border?: boolean;
  like?: boolean;
  // onClick: (event: React.MouseEvent<HTMLDivElement>) => void;
}
