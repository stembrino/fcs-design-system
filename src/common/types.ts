export interface CSSProperties extends React.CSSProperties {
  "--focus-color"?: string;
  "--button-color"?: string;
  "--border-color"?: string;
}
