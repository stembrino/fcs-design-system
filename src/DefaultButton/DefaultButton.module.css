:root {
  --border-color: #000000;
  --button-primary: #000000;
}

.test-component {
  background-color: var(--border-color);
  color: var(--font-color);
  border: 1px solid var(--harvey-black);
  padding: 16px;
  width: 360px;
  text-align: center;
}

.test-component .heading {
  font-size: var(--heading-font-size);
}
