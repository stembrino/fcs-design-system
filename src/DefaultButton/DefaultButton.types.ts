export interface DefaultButtonProps {
  label: string;
}
