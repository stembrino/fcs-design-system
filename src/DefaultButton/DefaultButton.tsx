import React from "react";
import { DefaultButtonProps } from "./DefaultButton.types";
import "./DefaultButton.module.css";

const DefaultButton: React.FC<DefaultButtonProps> = ({ label }) => {
  return <div role="button">{label}</div>;
};

export default DefaultButton;
