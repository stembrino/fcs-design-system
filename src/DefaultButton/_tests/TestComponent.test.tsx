import React from "react";
import { render, screen } from "@testing-library/react";

import DefaultButton from "../.";
import { DefaultButtonProps } from "../DefaultButton.types";

describe("TestComponent", () => {
  const renderComponent = ({ label = "test" }: Partial<DefaultButtonProps>) => {
    const { container } = render(<DefaultButton label={label} />);

    return { container };
  };

  it("should render heading text correctly", () => {
    renderComponent({});

    expect(screen).toBeInTheDocument();
  });
});
