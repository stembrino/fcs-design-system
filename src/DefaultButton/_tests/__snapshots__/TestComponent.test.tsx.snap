// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render heading text correctly 1`] = `
<div>
  <div
    role="button"
  >
    test
  </div>
</div>
`;
